android/modules/network/src/java/ti/modules/titanium/network/socket/TCPProxy.java

```diff
@@ -220,22 +233,46 @@ public void run()
 			Object timeoutProperty = getProperty("timeout");
 
 			try {
-				if (timeoutProperty != null) {
-					int timeout = TiConvert.toInt(timeoutProperty, 0);
-
+				int port = TiConvert.toInt(getProperty("port"));
+				boolean startTls = TiConvert.toBoolean(getProperty("startTls"), false);
+
+				if (startTls) {
+					SocketFactory sslSocketFactory = SSLSocketFactory.getDefault();
+					SSLSocket sslSocket = (SSLSocket) sslSocketFactory.createSocket();
+					sslSocket.setUseClientMode(true);
+					SSLParameters sslParameters = new SSLParameters();
+					List sniHostNames = new ArrayList(1);
+					sniHostNames.add(new SNIHostName(host));
+					sslParameters.setServerNames(sniHostNames);
```

Comment:
The `SNIHostName` class and `SSLParameters.setServerNames()` method are API Level 24 only (aka: Android 7.0).
https://developer.android.com/reference/javax/net/ssl/SNIHostName
https://developer.android.com/reference/javax/net/ssl/SSLParameters#setServerNames(java.util.List%3Cjavax.net.ssl.SNIServerName%3E)

This means this code will fail on Android OS versions older than 7.0. Our min supported API Level is 19 (aka: Android 4.4). So, this code will need to be rewritten to work on older OS versions. If we can't find an alternative solution, then we'll have to guard the code as follows...
```java
if (Build.VERSION.SDK_INT >= 24) {
	SSLParameters sslParameters = new SSLParameters();
	List sniHostNames = new ArrayList(1);
	sslParameters.setServerNames(sniHostNames);
	sslSocket.setSSLParameters(sslParameters);
}
```

I suggest testing this with an Android 4.4 device/emulator to verify that your code changes works.